repair/repair.cc

```diff
@@ -1125,16 +1139,25 @@ future<> tasks::data_sync_repair_task_impl::run() {
             throw std::runtime_error("aborted by user request");
         }
         for (auto shard : boost::irange(unsigned(0), smp::count)) {
-            auto f = rs.container().invoke_on(shard, [this, keyspace, table_ids, id, ranges, neighbors, reason, ops_uuid] (repair_service& local_repair) mutable {
-                auto data_centers = std::vector<sstring>();
-                auto hosts = std::vector<sstring>();
-                auto ignore_nodes = std::unordered_set<gms::inet_address>();
-                bool hints_batchlog_flushed = false;
-                auto ri = make_lw_shared<repair_info>(local_repair,
-                        std::move(keyspace), std::move(ranges), std::move(table_ids),
-                        id, std::move(data_centers), std::move(hosts), std::move(ignore_nodes), reason, ops_uuid, hints_batchlog_flushed);
-                ri->neighbors = std::move(neighbors);
-                return repair_ranges(ri);
+            auto f = rs.container().invoke_on(shard, [keyspace, table_ids, id, ranges, neighbors, reason, ops_uuid, parent_data = get_repair_uniq_id().task_data] (repair_service& local_repair) mutable -> future<> {
+                lw_shared_ptr<repair_info> ri;
+                std::exception_ptr ex;
+                auto task_id = tasks::task_id{id.uuid().uuid()};
+                try {
+                    auto data_centers = std::vector<sstring>();
+                    auto hosts = std::vector<sstring>();
+                    auto ignore_nodes = std::unordered_set<gms::inet_address>();
+                    bool hints_batchlog_flushed = false;
+                    auto ri = make_lw_shared<repair_info>(local_repair,
+                            std::move(keyspace), std::move(ranges), std::move(table_ids),
+                            id, std::move(data_centers), std::move(hosts), std::move(ignore_nodes), reason, ops_uuid, hints_batchlog_flushed);
+                    ri->neighbors = std::move(neighbors);
+                } catch (...) {
+                    ex = std::current_exception();
+                }
+                auto task_impl_ptr = std::make_unique<tasks::shard_repair_task_impl>(local_repair._repair_module, task_id::create_random_id(), 0, keyspace, "", format("{}", streaming::stream_reason::repair), "", task_id, ri, ex);
+                auto task = co_await start_repair_task(std::move(task_impl_ptr), local_repair._repair_module, parent_data);
```

Comment:
We don't need to do `task->start()` here?

Reply:
`start_repair_task` does it.

Reply:
I see.